ec/ecresource/deploymentresource/deployment/v2/deployment_update_payload.go

```diff
@@ -60,48 +63,63 @@ func (plan DeploymentTF) getBaseUpdatePayloads(ctx context.Context, client *api.
 		Kibana:             template.DeploymentTemplate.Resources.Kibana,
 	}
 
-	// If the deployment template has changed then we should use the template migration API
-	// to build the base update payloads
-	if newDtId != prevDtId && prevDtId != "" {
-		// Get an update request from the template migration API
-		migrateUpdateRequest, err := client.V1API.Deployments.MigrateDeploymentTemplate(
-			deployments.NewMigrateDeploymentTemplateParams().WithDeploymentID(plan.Id.ValueString()).WithTemplateID(newDtId),
-			client.AuthWriter,
-		)
+	planHasNodeTypes, diags := elasticsearchv2.PlanHasNodeTypes(ctx, plan.Elasticsearch)
+
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	templateChanged := newDtId != prevDtId && prevDtId != ""
+
+	// If the deployment template has changed or MigrateToLatestHardware is true, we should use the template migration
+	// API to build the base update payloads
+	migrateToLatest := plan.MigrateToLatestHardware.ValueBool() || templateChanged
+
+	// Template migration isn't available for deployments using node types
+	if migrateToLatest && !planHasNodeTypes {
+		// If the template has changed, we can't use the migrate request from private state.
+		// In this case, we fetch a new update request again from the template migration API
+		if migrateTemplateRequest == nil {
```

Comment:
I guess we cannot use `migrateTemplateRequest` if `templateChanged` is true, can we?

Reply:
Shall we change this to
```suggestion
		if migrateTemplateRequest == nil || templateChanged {
```

Reply:
I intended to do this, I added a comment about it and everything but then forgot to add the clause 😅 thanks for pointing it out! Added in https://github.com/elastic/terraform-provider-ec/pull/755/commits/a69026c9d9ed2acf9c0f3872779d6be75fb05bb6.
The strange thing is that the acceptance tests worked without it 🤔